base/gmp.jl

```diff
@@ -553,6 +554,7 @@ binomial(n::BigInt, k::Integer) = k < 0 ? BigInt(0) : binomial(n, UInt(k))
 ==(x::BigInt, f::CdoubleMax) = isnan(f) ? false : cmp(x,f) == 0
 ==(f::CdoubleMax, x::BigInt) = isnan(f) ? false : cmp(x,f) == 0
 iszero(x::BigInt) = x.size == 0
+isone(x::BigInt) = x == ONE
```

Comment:
This is correct, but this uses comparison between two `BigInt`: I wonder if it wouldn't be slightly faster to test `x == Clong(1)`? 
Ok I tested: on my machine it's about 4.1ns (`Clong` approach) vs 4.6ns (`ONE`), and 3.8ns when comparing with `Culong(1)`. Not sure we should bother for such small differences, but it doesn't cost much here to choose the fastest, can you reproduce those benchmarks to confirm?